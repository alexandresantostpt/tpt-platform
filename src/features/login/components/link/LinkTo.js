import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import { colors } from '@helpers/colors'

import { Link } from 'react-router-dom'

const StyledLink = styled(Link)`
    color: ${colors.link};
`

const LinkTo = ({ text, to }) => <StyledLink to={to}>{text}</StyledLink>

LinkTo.propTypes = {
    text: PropTypes.string.isRequired,
    to: PropTypes.string.isRequired
}

export default LinkTo
