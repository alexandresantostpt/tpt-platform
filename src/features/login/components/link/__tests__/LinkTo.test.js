import React from 'react'
import renderer from 'react-test-renderer'

import LinkTo from '../LinkTo'

describe('Tests for LinkTo component', () => {
    it('Should rendering without crash', () => {
        const tree = renderer.create(<LinkTo text="123" to="/" />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
