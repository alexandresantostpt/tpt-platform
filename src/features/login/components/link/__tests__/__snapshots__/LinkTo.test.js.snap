// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for LinkTo component Should rendering without crash 1`] = `
<a
  className="LinkTo__StyledLink-sc-1u70co-0 khxOGt"
  to="/"
>
  123
</a>
`;
